### PR TITLE
delay project system init to avoid solution update race

### PR DESCRIPTION
**Pull request by corngood**
I've been getting this error intermittently:

https://github.com/OmniSharp/omnisharp-roslyn/blob/8f3b6ea9915c4e692ca324a1e2e58e394a6cac75/src/OmniSharp.MSBuild/ProjectManager.cs#L375

I tracked it down to a race between `TryApplyChanges` calls.

Adding the project happens on a worker thread, and is queued on the main thread inside:

https://github.com/OmniSharp/omnisharp-roslyn/blob/8f3b6ea9915c4e692ca324a1e2e58e394a6cac75/src/OmniSharp.Host/WorkspaceInitializer.cs#L43

I believe `AddProject` can be called on a worker thread any time after this.

The other call is on the main thread via:

https://github.com/OmniSharp/omnisharp-roslyn/blob/8f3b6ea9915c4e692ca324a1e2e58e394a6cac75/src/OmniSharp.Host/WorkspaceInitializer.cs#L59

https://github.com/OmniSharp/omnisharp-roslyn/blob/8f3b6ea9915c4e692ca324a1e2e58e394a6cac75/src/OmniSharp.Host/WorkspaceInitializer.cs#L91

This change rearranges workspace init to avoid this particular race.  It doesn't seem like anything attempts to retry solution changes, so if there are other races it may be a more serious problem.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-roslyn?pullRequest=2057) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-roslyn?pullRequest=2057) before we can accept your contribution.<br/><br/>:x: corngood [sign now](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-roslyn?pullRequest=2057)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/OmniSharp/omnisharp-roslyn?pullRequest=2057) it.</sub>

File: src/OmniSharp.Host/WorkspaceInitializer.cs [hunk 1]
```diff
@@ -27,11 +27,25 @@ public static void Initialize(IServiceProvider serviceProvider, CompositionHost
 
             var projectEventForwarder = compositionHost.GetExport<ProjectEventForwarder>();
             projectEventForwarder.Initialize();
-            var projectSystems = compositionHost.GetExports<IProjectSystem>();
 
             workspace.EditorConfigEnabled = options.CurrentValue.FormattingOptions.EnableEditorConfigSupport;
             options.OnChange(x => workspace.EditorConfigEnabled = x.FormattingOptions.EnableEditorConfigSupport);
 
+            logger.LogDebug("Starting with OmniSharp options: {options}", options.CurrentValue);
+            ProvideWorkspaceOptions(compositionHost, workspace, options, logger, omnisharpEnvironment);
```
Comment (filipw) [inline src/OmniSharp.Host/WorkspaceInitializer.cs:35]: thanks, I think this is OK to change, however the `workspace.Initialized = true;` needs to be moved back down until after the project systems run their inits

Comment (corngood) [reply src/OmniSharp.Host/WorkspaceInitializer.cs:35]: Oops, good catch.  Fixed

**Review by filipw — APPROVED** [review]
LGTM

**Comment by filipw** [comment]
@JoeRobich @david-driscoll @mholo65 do you have any comments on this?

**Review by JoeRobich — APPROVED** [review]

**Comment by filipw** [comment]
thanks!

**Pull request merged**